File: frontend/src/api/medication.ts
import api from './client'

export interface Medication {
  id: number
  elder_id: number
  drug_name: string
  generic_name?: string
  dosage: string
  frequency: string
  route: string
  start_date?: string
  end_date?: string
  prescribing_doctor?: string
  pharmacy?: string
  purpose?: string
  side_effects?: string
  notes?: string
  status: string
  created_at: string
  updated_at: string
}

export const medicationApi = {
  getList: (elderId: number) => {
    return api.get<Medication[]>(`/elders/${elderId}/medications`)
  },

  create: (elderId: number, data: Partial<Medication>) => {
    return api.post<Medication>(`/elders/${elderId}/medications`, data)
  },

  update: (id: number, data: Partial<Medication>) => {
    return api.put<Medication>(`/medications/${id}`, data)
  },

  updateStatus: (id: number, status: string) => {
    return api.patch<Medication>(`/medications/${id}/status`, { status })
  },

  remove: (id: number) => {
    return api.delete(`/medications/${id}`)
  },
}

File: backend/app/models/medication.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text, Date, Time, Boolean
from sqlalchemy.orm import relationship

from app.database import Base


class Medication(Base):
    __tablename__ = "medications"

    id = Column(Integer, primary_key=True, index=True)
    elder_id = Column(Integer, ForeignKey("elders.id"), nullable=False, index=True)
    drug_name = Column(String(200), nullable=False)
    generic_name = Column(String(200), nullable=True)
    dosage = Column(String(100), nullable=False)
    frequency = Column(String(100), nullable=False)
    route = Column(String(50), nullable=False)
    start_date = Column(Date, nullable=True)
    end_date = Column(Date, nullable=True)
    prescribing_doctor = Column(String(100), nullable=True)
    pharmacy = Column(String(200), nullable=True)
    purpose = Column(Text, nullable=True)
    side_effects = Column(Text, nullable=True)
    notes = Column(Text, nullable=True)
    status = Column(String(20), nullable=False, default="active")
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    elder = relationship("Elder", back_populates="medications")
